### DISCUSSION TITLE
Notifications and session

### DISCUSSION THREAD

**Discussion by bernhardh** [wireui/wireui#149] [Ideas]
I think a common usecase for notifications is to show a success on save of a form. This works as expected, as long as there is no url change.

There should be (or am I missing something?) a build in way to achive this, maybe with `Session` (flash?)

**Comment by bernhardh** [comment]
I still think, this could be a build in function, but I implemented it now, just with a minor issue (#150).

I created a helper function `notify` (in `app/helper.php`:

```php
function notify(string $title, string $msg = "", string $type = "success") {
    session()->push("wireui.notify", [
        "title" => $title,
        "description" => $msg,
        "icon" => $type
    ]);
}
```
In the (layout) template I added:
```html
@if($toasts = session()->pull("wireui.notify", []))
    <script>
        Wireui.hook('notifications:load', () => {
            setTimeout(function (){
                @foreach($toasts AS $toast) window.$wireui.notify({!! json_encode($toast) !!}); @endforeach
            }, 200);
        })
    </script>
@endif
```
and now I can call from the controller or elsewhere:

```php
notify("Success");
notify("Error", "Some more infos", "error");
```


Not sure if this is the best solution, but it works.

**Comment by ph7jack** [comment]
@bernhardh It's a good idea. I will soon bring many improvements and bug fixes, I'm now on a vacation

**Comment by leolll** [comment]
@bernhardh I took your code and modified it so you can use `notify()` anywhere in your code and a notification will show when a livewire component is re-rendered (instead of a page reload).  I had to add a `renderingActions()` method to the `Actions` trait:

```php
// Added to custom WireUi\Traits\Actions trait
public function renderingActions()
{
    if ($notifications = session()->pull('wireui.notify')) {
        foreach ($notifications as $notification) {
            $this->notification()->notify([
                'title'       => $notification['title'],
                'description' => $notification['description'],
                'icon'        => $notification['icon'],
            ]);
        }
    }
}
```

@PH7-Jack Fantastic package!  Do you think this approach has any downsides?

**Reply by ph7jack** [reply]
@leolll Yes, it's a good way. I soon it will be available in the wireui

**Comment by Krato** [comment]
Is there some update on this? Thanks
